src/ApiCall.php

```diff
@@ -105,41 +112,70 @@ public function get(string $endPoint, array $params, bool $asJson = true)
      * @param mixed $body
      *
      * @param bool $asJson
+     * @param array $queryParameters
      *
      * @return array|string
      * @throws TypesenseClientError
      * @throws GuzzleException
      */
-    public function post(string $endPoint, $body, bool $asJson = true)
+    public function post(string $endPoint, $body, bool $asJson = true, array $queryParameters = [])
     {
         return $this->makeRequest('post', $endPoint, $asJson, [
             'data' => $body ?? [],
+            'query' => $queryParameters ?? []
         ]);
     }
 
     /**
      * @param string $endPoint
      * @param array $body
      *
+     * @param bool $asJson
+     * @param array $queryParameters
+     *
      * @return array
      * @throws TypesenseClientError|GuzzleException
      */
-    public function put(string $endPoint, array $body): array
+    public function put(string $endPoint, array $body, bool $asJson = true, array $queryParameters = []): array
     {
-        return $this->makeRequest('put', $endPoint, true, [
+        return $this->makeRequest('put', $endPoint, $asJson, [
             'data' => $body ?? [],
+            'query' => $queryParameters ?? []
         ]);
     }
 
     /**
      * @param string $endPoint
+     * @param array $body
+     *
+     * @param bool $asJson
+     * @param array $queryParameters
      *
      * @return array
      * @throws TypesenseClientError|GuzzleException
      */
-    public function delete(string $endPoint): array
```

Comment:
Why removing the delete method for patch instead of adding patch as a new method.

Reply:
The diff makes it look like I removed it, but I actually just _moved_ `delete` from further down in the file to be close to the other http methods in `ApiCall`:

https://github.com/typesense/typesense-php/pull/3/files#diff-198237dfe5521226a362027889383cb3294ed90976c8de7190dcf0e15f1bb3a1R174-R178